python/_tskitmodule.c

```diff
@@ -779,6 +779,31 @@ uint32_converter(PyObject *py_obj, uint32_t *uint_out)
     return ret;
 }
 
+static int
+tsk_id_converter(PyObject *py_obj, tsk_id_t *id_out)
+{
+    long long temp_long;
+    int ret = 0;
+
+    if (!PyArg_Parse(py_obj, "L", &temp_long)) {
+        goto out;
+    }
+    if (temp_long > TSK_MAX_ID) {
+        PyErr_SetString(PyExc_OverflowError, "Value too large for tskit id type");
+        goto out;
+    }
+    if (temp_long < TSK_NULL) {
```

Comment:
Not sure this is worth checking at this level, but I don't feel strongly about it.

Reply:
I've done this this way as it should cope with both 32 and 64 bit ids should we wish to support that in future.